swagger_server/jobs/pull_topo_changes.py

```diff
@@ -0,0 +1,86 @@
+import argparse
+import json
+import logging
+import os.path
+import sys
+import threading
+import time
+import urllib.request
+
+sys.path.append(
+    os.path.abspath(os.path.join(os.path.dirname(__file__), os.path.pardir))
+)
+
+from messaging.rpc_queue_producer import *
+from utils.db_utils import *
+
+DOMAIN_CONTROLLER_PULL_URL = os.environ.get("DOMAIN_CONTROLLER_PULL_URL")
+DOMAIN_CONTROLLER_PULL_INTERVAL = os.environ.get("DOMAIN_CONTROLLER_PULL_INTERVAL")
+logger = logging.getLogger(__name__)
+
+
+def main():
+    db_instance = DbUtils()
+    db_instance.initialize_db()
+
+    process_domain_controller_topo(db_instance)
+
+
+def process_domain_controller_topo(db_instance):
+    while True:
+        latest_topology_exists = False
+        latest_topology = db_instance.read_from_db("latest_topology")
+
+        if latest_topology:
+            latest_topology_exists = True
+            json_latest_topology = json.loads(latest_topology["latest_topology"])
+
+            try:
+                latest_topo_version = json_latest_topology["version"]
+            except KeyError:
+                logger.debug("Error getting topo version")
+                continue
+        else:
+            logger.debug("Latest topology does not exist")
+
+        try:
+            pulled_topology = urllib.request.urlopen(DOMAIN_CONTROLLER_PULL_URL).read()
+        except urllib.request.URLError:
+            logger.debug("Error connecting to domain controller...")
+            time.sleep(int(DOMAIN_CONTROLLER_PULL_INTERVAL))
+            continue
+
+        if not pulled_topology:
+            time.sleep(int(DOMAIN_CONTROLLER_PULL_INTERVAL))
+            continue
+
+        logger.debug("Pulled request from domain controller")
+
+        json_pulled_topology = json.loads(pulled_topology)
```

Comment:
Seems that we're assuming that `pulled_topology` is always JSON, but this assumption can go wrong. What happens if `pulled_topology` is not JSON, but maybe an error message (40x/50x)?

Reply:
Yeah great catch. Since this is not inside an API endpoint, maybe we can just ignore the pulled topology if it's not a valid JSON. I moved it to a try/catch block.